bmap-rs/src/main.rs

```diff
@@ -89,12 +156,43 @@ fn setup_input(path: &Path) -> Result<Decoder> {
     }
 }
 
-fn copy(c: Copy) -> Result<()> {
-    if !c.image.exists() {
-        bail!("Image file doesn't exist")
+async fn setup_remote_input(url: Url) -> Result<Response> {
+    if PathBuf::from(url.path()).extension().unwrap() != "gz" {
+        bail!("Image file format not implemented")
     }
+    reqwest::get(url).await.map_err(anyhow::Error::new)
+}
+
+fn setup_progress_bar(bmap: &Bmap) -> Result<ProgressBar> {
+    let pb = ProgressBar::new(bmap.total_mapped_size());
+    pb.set_style(ProgressStyle::with_template("{spinner:.green} [{elapsed_precise}] [{wide_bar:.cyan/blue}] {bytes}/{total_bytes} ({eta})")
+        .unwrap()
+        .with_key("eta", |state: &ProgressState, w: &mut dyn Write| write!(w, "{:.1}s", state.eta().as_secs_f64()).unwrap())
+        .progress_chars("#>-"));
+    Ok(pb)
```

Comment:
It doesn't seem like anything here actually uses the Err side? So it could just return it w/o the Result wrapper.